src/external-server/docker-server.ts

```diff
@@ -46,9 +47,15 @@ export abstract class DockerServer extends ExternalServer {
             formattedImage,
             ...(await this.getContainerArgs())
         ];
+
         return spawn("docker", args);
     }
 
+    public isDockerDesktop(): boolean {
```

Comment:
I would implement this as a cached [JS getter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) (does it have to be public?) which first checks a private property to see if it was already executed.

Reply:
its only executed once when spawning docker container